src/main/java/com/github/mmolimar/kafka/connect/fs/util/BatchIterator.java

```diff
@@ -0,0 +1,29 @@
+package com.github.mmolimar.kafka.connect.fs.util;
+
+import java.util.Iterator;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BatchIterator {
```

Comment:
We could add this functionality inside the ``listFiles`` method in the ``AbstractPolicy`` class

Reply:
If we put it in the `listFiles` method we will be batching each file system independently. Feels to me that is not the intended behaviour.

Reply:
Ok. Maybe renaming this class to ``Iterators`` with a ``partition`` method with the batch size and a "duplicates" flag to allow removing duplicates in the same partition.

Reply:
Sorry, I didn't understand what you mean by "allow removing duplicates in the same partition".

Reply:
For example: in the first execution of the policy you get an iterator with 5 records (with a batch size of 3) so 2 remaining. In the next iteration (let's say we're using the Cron policy) we concat the previous 2 with other 5 records. We re-split that iterator and "maybe" in this batch could be an item duplicated, so we should ignore this record and get another one.

Reply:
I don't think that will not be inline with the present implementation. 
An execution is only finished when the iterator is fully traversed, only after that we will sleep.  If we mix files from the current iterator with the next one we will be creating a weird behaviour in the policies.